libs/payments/stripe/src/lib/stripe.manager.ts

```diff
@@ -100,14 +101,43 @@ export class StripeManager {
   }
 
   async retrieveSubscription(subscriptionId: string) {
-    return this.client.subscriptionsRetrieve(subscriptionId);
+    try {
+      const subscription = await this.client.subscriptionsRetrieve(
+        subscriptionId
+      );
+      return subscription;
+    } catch (err) {
+      if (err.type === 'StripeInvalidRequestError') {
+        throw new PromotionCodeCouldNotBeAttachedError(
+          'Subscription not found',
+          err
+        );
```

Comment:
Similar to above, these all need to be moved up to the service layer. The manager shouldn't throw errors related to a service operation.